querybook/server/clients/github_client.py

```diff
@@ -0,0 +1,130 @@
+from flask import session as flask_session
+from github import Github, GithubException, Auth
+from typing import List, Dict, Optional
+
+from lib.github_integration.serializers import (
+    deserialize_datadoc_from_markdown,
+    serialize_datadoc_to_markdown,
+)
+from lib.logger import get_logger
+from models.datadoc import DataDoc
+from models.github import GitHubLink
+from env import QuerybookSettings
+
+LOG = get_logger(__name__)
+
+
+class GitHubClient:
+    def __init__(self, github_link: GitHubLink, datadoc: DataDoc):
+        """
+        Initialize the GitHub client with an access token from the session.
+        Raises an exception if the token is not found.
+        """
+        access_token = self._get_access_token()
+        auth = Auth.Token(access_token)
+        self.client = Github(auth=auth)
+        self.user = self.client.get_user()
+        self.github_link = github_link
+        self.repo = self._get_repository()
+        self.branch = "main"
+        self.file_path = self._build_file_path(datadoc)
+
+    def _get_access_token(self) -> str:
+        access_token = flask_session.get("github_access_token")
+        if not access_token:
+            LOG.error("GitHub OAuth token not found in session")
+            raise Exception("GitHub OAuth token not found in session")
+        return access_token
+
+    def _get_repository(self):
+        repo_url = QuerybookSettings.GITHUB_REPO_URL
+        if not repo_url:
+            LOG.error("GITHUB_REPO_URL is not configured")
+            raise Exception("GITHUB_REPO_URL is not configured")
+        repo_full_name = self._extract_repo_full_name(repo_url)
```

Comment:
if repo_url is only used to parse the repo full name, why not just provide the repo name in the config directly?